File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/renderer/AccountRenderer.java
package de.fu_berlin.inf.dpp.ui.renderer;

import de.fu_berlin.inf.ag_se.browser.extensions.IJQueryBrowser;
import de.fu_berlin.inf.dpp.HTMLUIContextFactory;
import de.fu_berlin.inf.dpp.account.IAccountStoreListener;
import de.fu_berlin.inf.dpp.account.XMPPAccount;
import de.fu_berlin.inf.dpp.account.XMPPAccountStore;
import de.fu_berlin.inf.dpp.ui.JavaScriptAPI;
import java.util.ArrayList;
import java.util.List;

/** This class is responsible for sending the account list to the HTML UI. */
public class AccountRenderer extends Renderer {

  private List<XMPPAccount> accounts;

  /**
   * Created by PicoContainer
   *
   * @param accountStore
   * @see HTMLUIContextFactory
   */
  public AccountRenderer(XMPPAccountStore accountStore) {
    this.accounts = new ArrayList<XMPPAccount>();

    accountStore.addListener(
        new IAccountStoreListener() {
          @Override
          public void accountsChanged(List<XMPPAccount> currentAccounts) {
            update(currentAccounts);
            render();
          }
        });
  }

  private synchronized void update(List<XMPPAccount> allAccounts) {
    accounts = new ArrayList<XMPPAccount>(allAccounts);
  }

  @Override
  public synchronized void render(IJQueryBrowser browser) {
    JavaScriptAPI.updateAccounts(browser, accounts);
  }
}

File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/renderer/AccountRenderer.kt
package de.fu_berlin.inf.dpp.ui.renderer

import java.util.ArrayList
import de.fu_berlin.inf.ag_se.browser.extensions.IJQueryBrowser
import de.fu_berlin.inf.dpp.HTMLUIContextFactory
import de.fu_berlin.inf.dpp.account.NullAccountStoreListener
import de.fu_berlin.inf.dpp.account.XMPPAccount
import de.fu_berlin.inf.dpp.account.XMPPAccountStore
import de.fu_berlin.inf.dpp.ui.JavaScriptAPI

/**
 * This class is responsible for sending the account list to the HTML UI.
 */
class AccountRenderer
/**
 * Created by PicoContainer
 *
 * @param accountStore
 * @see HTMLUIContextFactory
 */
	(accountStore: XMPPAccountStore?) : Renderer() {
	private var accounts: List<XMPPAccount>? = ArrayList<XMPPAccount>()

	init {
		accountStore!!.addListener(object : NullAccountStoreListener() {
			override fun accountsChanged(currentAccounts: List<XMPPAccount>?) {
				update(currentAccounts)
				render()
			}
		})
	}

	@Synchronized
	private fun update(allAccounts: List<XMPPAccount>?) {
		accounts = ArrayList<XMPPAccount>(allAccounts)
	}

	
	@Synchronized
	public override fun render(browser: IJQueryBrowser?) {
		JavaScriptAPI.updateAccounts(browser, accounts)
	}
}